src/pages/EditorPage.js

```diff
@@ -339,28 +365,166 @@ export default function EditorPage(props) {
     [setLayout, tab, setTab]
   );
 
-  const widgetName = path?.name;
+  const widgetName = path?.name.split("/")[0];
+  const widgetPathName = path?.name;
+  const isDraft = path?.name.split("/")[2] === "draft";
 
-  const commitButton = (
-    <CommitButton
+  const widgetPath = `${accountId}/${path?.type}/${path?.name}`;
+  const jpath = JSON.stringify(path);
+
+  const createOpenDraftButton = (
+    <button
       className="btn btn-primary"
+      onClick={(e) => {
+        openDraft(widgetName);
+      }}
+    >
+      Open a Draft version
```

Comment:
What's the expected behavior here? I clicked this button while I had the editor open showing the code of this component `charleslavon.near/widget/ActivityPageMinusExplore` and it opened a tab titled `ActivityPageMinusExplore DRAFT` but instead of copying the code it displayed ```return <div>Hello World</div>;```

Reply:
Current implementation fetches the draft version from chain. If the draft version is not present on chain, the code defaults to "Hello World". We could default to the code a draft is created from and this may seem a bit more intuitive. 

My concern here is how to treat the version saved to localStorage. If we have both a draft version on chain and a version saved locally, how should be decided which version to pull. Forcing the user to save locally before pushing as draft seems to solve this issue (see: https://github.com/near/near-discovery-alpha/pull/2#discussion_r1114874963).

Reply:
How do you propose forcing a user to save locally before saving a draft? Does the `Publish Draft to Main` button only appear if a draft version of this component exists?

Reply:
I think that `Open a Draft version` should open a new tab with the code from the main component - local code. 

Additionally, I would propose adding a button called `Revert to on-chain version` to revert the component code, to the version that exists on chain. So as a result the local version will be replaced with the on-chain version.

Reply:
@charleslavon we may either compare the code version with the version saved locally and allow only if the two match (by disabling a button and showing a tooltip for example). Second option is to always save locally when publishing on chain but this may seem rather confusing for a user.

@marcinbodnar @charleslavon I modified `Open a Draft Version` button to copy the code from the main version.